mat64/triangular.go

```diff
@@ -115,7 +115,18 @@ func (t *TriDense) Dims() (r, c int) {
 }
 
 func (t *TriDense) Triangle() (n int, upper bool) {
-	return t.mat.N, t.mat.Uplo == blas.Upper
+	return t.mat.N, t.isUpper()
```

Comment:
In Cholesky the TriDense's uplo is not queried (I should check for other cases wher the value uplo == Upper is used). The only thing that doesn't work after this is calling Triangle on the zero value AFAICS.